docs/dev/graphics-stack.md

```diff
@@ -0,0 +1,398 @@
+<!--
+
+The contents of this Documentation are subject to the Public Documentation License Version 1.01
+(the "License"); you may only use this Documentation if you comply with the terms of this License.
+A copy of the License is available at http://illumos.org/license/PDL.
+
+The Original Documentation is _________________.
+
+The Initial Writer of the Original Documentation is Aurelien Larcher 2017.
+All Rights Reserved.
+
+Portions created by Ken Mays are Copyright (C) 2017.
+Portions created by Alexander Pyhalov are Copyright (C) 2019.
+
+-->
+
+<img src = "../../Openindiana.png">
+
+# Graphics stack
+
+Like most UNIX/Linux operating systems, OpenIndiana's GUI environment is based on
+the X Window System bundled with libraries and applications developed by
+the [freedesktop.org](https://www.freedesktop.org/) community.
+
+Therefore the Graphic Stack is comprised of:
+* the opensource X11 display server Xorg, client libraries and utilities developed by the [X.org Project](https://www.x.org/),
+* an OpenGL implementation in the form of the opensource [Mesa](https://www.mesa3d.org/) library,
+* and illumos kernel drivers for different graphics adapters provided by the [gfx-drm](https://github.com/illumos/gfx-drm) gate,
+* together with the [libdrm](https://cgit.freedesktop.org/mesa/drm) implementing communication between kernel drivers and user-space components through the Direct Rendering Infrastructure (DRI) protocol.
+
+All these components can be installed using the unified build system [oi-userland](https://github.com/OpenIndiana/oi-userland) or simply installed from the package repositories.
+
+Notes concerning different components are provided:
+
+ Component             | Scope
+ --------------------- | ----------------
+   [Xorg](#Xorg)       | Building the Xorg display server, protocol headers, client libraries and applications.
+   [Mesa](#Mesa)       | Building the Mesa library and testing OpenGL capabilities.
+   [Fonts](#Fonts)     | Packaging new fonts for OpenIndiana.
+   [Nvidia](#Nvidia)   | Installation of Nvidia's proprietary graphics driver for Solaris.
+   [gfx-drm](#gfx-drm) | Building the illumos KMS drivers and libdrm from the gfx-drm gate.
+
+## Xorg
+
+All the Xorg components are build in oi-userland and located in the x11 directory:
+
+<https://github.com/OpenIndiana/oi-userland/tree/oi/hipster/components/x11>
+
+A bogus component is provided to rebuild the X11 gate:
+<https://github.com/OpenIndiana/oi-userland/tree/oi/hipster/components/x11/x11-gate>
+
+The upstream Solaris gate is located at:
+
+<https://github.com/oracle/solaris-xorg/>
```

Comment:
Certainly dead? The page says "This repository has been archived by the owner. It is now read-only."